core/metacling/src/TCling.cxx

```diff
@@ -1537,8 +1537,7 @@ TCling::TCling(const char *name, const char *title, const char* const argv[], vo
                                                        interpLibHandle);
 
    if (!fInterpreter->getCI()) { // Compiler instance could not be created. See https://its.cern.ch/jira/browse/ROOT-10239
-      std::cerr << "Exiting now since compiler instance is not available." << std::endl;
-      exit(EXIT_FAILURE);
+      return;
```

Comment:
Yes, I added safety for that in several places.

Reply:
Since `TCling` does not have a `IsValid` function (unless I missed it under a different spelling), the user can not check whether it was properly setup.  I recommend throwing an exception here.

Reply:
This option was discussed in the previous PR. We discarded throwing it because it autoprinted a long stacktrace dump, which was what was tried to avoid on first place.

Reply:
I've implemented a straightforward IsValid function so that the user can crosscheck.

Reply:
It **must** be added be added to the abstract interface `core/meta/inc/TInterpreter.h` for the user to be able to access it (`TCling` is an internal only derived class from `TInterpreter`).